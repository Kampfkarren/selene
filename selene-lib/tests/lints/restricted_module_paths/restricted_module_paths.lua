-- Test local assignment restriction
local deprecatedFunction = OldLibrary.Utils.deprecatedFunction

-- Test function call restriction
OldLibrary.Utils.deprecatedFunction()

-- Test function argument restriction
fn(OldLibrary.Utils.deprecatedFunction)

-- Test table constructor restriction
local config = { callback = OldLibrary.Utils.deprecatedFunction }

-- Test return statement restriction
function getHandler()
    return OldLibrary.Utils.deprecatedFunction
end

-- Test nested table restriction
local nested = { deep = { handler = OldLibrary.Utils.deprecatedFunction } }

-- Test if expression restriction
local handler = condition and OldLibrary.Utils.deprecatedFunction or nil
