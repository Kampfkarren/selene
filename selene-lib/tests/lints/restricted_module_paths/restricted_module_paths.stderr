error[restricted_module_paths]: Module path `OldLibrary.Utils.deprecatedFunction` is restricted
  ┌─ restricted_module_paths.lua:2:28
  │
2 │ local deprecatedFunction = OldLibrary.Utils.deprecatedFunction
  │                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  │
  = OldLibrary.Utils.deprecatedFunction has been deprecated. Use NewLibrary.Utils.modernFunction instead.

error[restricted_module_paths]: Module path `OldLibrary.Utils.deprecatedFunction` is restricted
  ┌─ restricted_module_paths.lua:5:1
  │
5 │ OldLibrary.Utils.deprecatedFunction()
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  │
  = OldLibrary.Utils.deprecatedFunction has been deprecated. Use NewLibrary.Utils.modernFunction instead.

error[restricted_module_paths]: Module path `OldLibrary.Utils.deprecatedFunction` is restricted
  ┌─ restricted_module_paths.lua:8:4
  │
8 │ fn(OldLibrary.Utils.deprecatedFunction)
  │    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  │
  = OldLibrary.Utils.deprecatedFunction has been deprecated. Use NewLibrary.Utils.modernFunction instead.

error[restricted_module_paths]: Module path `OldLibrary.Utils.deprecatedFunction` is restricted
   ┌─ restricted_module_paths.lua:11:29
   │
11 │ local config = { callback = OldLibrary.Utils.deprecatedFunction }
   │                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │
   = OldLibrary.Utils.deprecatedFunction has been deprecated. Use NewLibrary.Utils.modernFunction instead.

error[restricted_module_paths]: Module path `OldLibrary.Utils.deprecatedFunction` is restricted
   ┌─ restricted_module_paths.lua:15:12
   │
15 │     return OldLibrary.Utils.deprecatedFunction
   │            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │
   = OldLibrary.Utils.deprecatedFunction has been deprecated. Use NewLibrary.Utils.modernFunction instead.

error[restricted_module_paths]: Module path `OldLibrary.Utils.deprecatedFunction` is restricted
   ┌─ restricted_module_paths.lua:19:37
   │
19 │ local nested = { deep = { handler = OldLibrary.Utils.deprecatedFunction } }
   │                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │
   = OldLibrary.Utils.deprecatedFunction has been deprecated. Use NewLibrary.Utils.modernFunction instead.

error[restricted_module_paths]: Module path `OldLibrary.Utils.deprecatedFunction` is restricted
   ┌─ restricted_module_paths.lua:22:31
   │
22 │ local handler = condition and OldLibrary.Utils.deprecatedFunction or nil
   │                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │
   = OldLibrary.Utils.deprecatedFunction has been deprecated. Use NewLibrary.Utils.modernFunction instead.

